netty/src/main/java/io/grpc/netty/NettyAdaptiveCumulator.java

```diff
@@ -23,6 +23,18 @@
 import io.netty.buffer.CompositeByteBuf;
 import io.netty.handler.codec.ByteToMessageDecoder.Cumulator;
 
+
+/**
+ * "Adaptive" cumulator: cumulate {@link ByteBuf}s by dynamically switching between merge and
+ * compose strategies.
+ * <br><br>
+ *
+ * <p><b><font color="red">Avoid using</font></b>
```

Comment:
I'm not sure this phrasing in red font makes a lot of sense here. It's not actionable for the user of this class. If it's for modifying the class itself, should we just note that on `mergeWithCompositeTail` method instead?

Reply:
It isn't just for `mergeWithCompositeTail`. It is for the entire class. So `cumulate()` and `addTail` are impacted as well.